homeassistant/components/enocean/light.py

```diff
@@ -44,46 +47,104 @@ def setup_platform(
     sender_id: list[int] = config[CONF_SENDER_ID]
     dev_name: str = config[CONF_NAME]
     dev_id: list[int] = config[CONF_ID]
-
-    add_entities([EnOceanLight(sender_id, dev_id, dev_name)])
+    brightness = config.get(ATTR_BRIGHTNESS)
+    color_mode = config.get(ATTR_COLOR_MODE)
+    if color_mode == ColorMode.ONOFF:
+        add_entities(
+            [
+                EnOceanLightNonDimmable(
+                    sender_id, dev_id, dev_name, brightness, color_mode
+                )
+            ]
+        )
+    else:
+        add_entities(
+            [EnOceanLightDimmable(sender_id, dev_id, dev_name, brightness, color_mode)]
+        )
 
 
 class EnOceanLight(EnOceanEntity, LightEntity):
     """Representation of an EnOcean light source."""
 
     _attr_color_mode = ColorMode.BRIGHTNESS
-    _attr_supported_color_modes = {ColorMode.BRIGHTNESS}
+    _attr_supported_color_modes = {ColorMode.BRIGHTNESS, ColorMode.ONOFF}
     _attr_brightness = 50
     _attr_is_on = False
 
-    def __init__(self, sender_id: list[int], dev_id: list[int], dev_name: str) -> None:
+    def __init__(
+        self,
+        sender_id: list[int],
+        dev_id: list[int],
+        dev_name: str,
+        brightness,
+        color_mode,
+    ) -> None:
         """Initialize the EnOcean light source."""
         super().__init__(dev_id)
         self._sender_id = sender_id
+        self._attr_brightness = brightness or self._attr_brightness
+        self._attr_color_mode = color_mode or self._attr_color_mode
+
         self._attr_unique_id = str(combine_hex(dev_id))
         self._attr_name = dev_name
 
     def turn_on(self, **kwargs: Any) -> None:
-        """Turn the light source on or sets a specific dimmer value."""
+        """Turn on light. Will send the provided command to Enocean relay."""
+        self.send(kwargs.get("command"))
+        self._attr_is_on = True
+
+    def turn_off(self, **kwargs: Any) -> None:
+        """Turn off light. Will send the provided command to Enocean relay."""
+        self.send(kwargs.get("command"))
+        self._attr_is_on = False
+
+    def send(self, command):
+        """Get the GFVS command, extend with 00 and append a sender_id before sending."""
+        command.extend(self._sender_id)
+        command.extend([0x00])
+        self.send_command(command, [], 0x01)
+
+
+class EnOceanLightNonDimmable(EnOceanLight):
+    """Representation of an EnOcean on/off light source."""
+
+    _attr_color_mode = ColorMode.ONOFF
+    _attr_supported_color_modes = {ColorMode.ONOFF}
+    _attr_is_on = False
+
+    def turn_on(self, **kwargs: Any) -> None:
+        """Turn the light source on. Simulate GFVS command."""
+        command = [0xA5, 0x01, 0x0, 0x00, 0x09]
```

Comment:
We won't accept more code that includes protocol details in this integration. Those should be extracted to a 3rd party library.